src/cryptoadvance/specter/services/swan/service.py

```diff
@@ -0,0 +1,223 @@
+            if details and "item" in details and "metadata" in details["item"] and "specter_wallet_alias" in details["item"]["metadata"]:
+                wallet_alias = details["item"]["metadata"]["specter_wallet_alias"]
+                if wallet_alias in app.specter.wallet_manager.wallets:
+                    # All is good; we've matched Swan's wallet data with a Specter `Wallet` that we recognize.
+                    if wallet_alias != service_data.get(SwanService.SPECTER_WALLET_ALIAS):
+                        cls.update_current_user_service_data({SwanService.SPECTER_WALLET_ALIAS: wallet_alias})
+                    return
+                else:
+                    # Swan is out of sync with Specter; the SPECTER_WALLET_ALIAS we had
+                    # been using doesn't exist on this Specter instance.
+                    # TODO: Alert the user and route them to settings to select a new Wallet?
+                    raise Exception(f"Swan configured to send to unknown wallet: {wallet_alias}.")
+            else:
+                # Specter's `swan_wallet_id` is out of sync; doesn't exist on Swan's side.
+                # Clear the local SPECTER_WALLET_ALIAS and continue below to try to find one.
+                del service_data[SwanService.SWAN_WALLET_ID]
+                cls.set_current_user_service_data(service_data)
+
+        # This Specter instance has no idea if there might already be wallet data on the Swan side.
+        # Fetch all Swan wallets, if any exist. 
+        wallet_entries = swan_api.get_wallets().get("list")
+        if not wallet_entries:
+            # No Swan data at all yet. Nothing to do.
+            return
+
+        swan_wallet_id = None
+        for wallet_entry in wallet_entries:
+            swan_wallet_id = wallet_entry["id"]
+            specter_wallet_alias = wallet_entry["metadata"].get("specter_wallet_alias")
+            if specter_wallet_alias in app.specter.wallet_manager.wallets:
+                # All is good; we've matched Swan's wallet data with a Specter `Wallet` that we recognize.
+                # Use this Swan walletId going forward.
+                cls.update_current_user_service_data({
+                    SwanService.SWAN_WALLET_ID: swan_wallet_id,
+                    SwanService.SPECTER_WALLET_ALIAS: specter_wallet_alias,
+                })
+                return
+        
+        # We didn't find a matching Specter `Wallet`. Clear out any nonsense settings in our local data.
+        if SwanService.SPECTER_WALLET_ALIAS in service_data:
+            del service_data[SwanService.SPECTER_WALLET_ALIAS]
+            SwanService.set_current_user_service_data(service_data)
+
+        # Did we at least get a Swan walletId that we can update later?
+        if swan_wallet_id:
+            cls.update_current_user_service_data({
+                SwanService.SWAN_WALLET_ID: swan_wallet_id,
+            })
+
+
+
+    @classmethod
+    def update(cls):
```

Comment:
Parent `Service` class does already define an empty placeholder. The temporary call in `redirect_login` is only there until a backend process is ready to take that task over.

As for the name, it shouldn't be `auth_` prefixed since the current `update()` call doesn't belong in there in the first place (again, it's only there because we don't have the backend process to call it).

I don't know that I'd call it a `callback`. Some process (the login flow for now, a backend process soon) has to tell each `Service` to update the state of its data (either remote 3rd party data or detect new local changes like a newly received utxo to an addr reserved for that `Service`). So `update` makes sense to me. But perhaps `sync` or `resync` if `update` really is the wrong name?

To be clear, I'm assuming the upcoming backend process would just do something like:
```
for service_cls in self.services:
    service_cls.update()
```
...repeatedly on some periodic basis.

Reply:
From a framework-perspective, it doesn't matter what the service is doing it's called regulary through some look like the one you described above. That's why i proposed a naming-scheme for such methods which rather describes when they got called, not what they are doing. 
If a service wants to update something, it can still define a  `_update()` method and internally call that in there function:
```
for service_cls in self.services:
    if hasAttr(service,"cron_callback_daily"): # Would not need the empty placeholder
        service_cls.cron_callback_daily()
```
And then in that method, one would call update:
```
    def cron_callback_daily():
        _update()
```

Does that make sense?

Reply:
Let's use hooknames like those:
* hooks called: on_user_login
* cron hooks like on_schedule_daily